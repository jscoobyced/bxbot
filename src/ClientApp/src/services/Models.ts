export enum Language {
    English,
    French
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
}

export interface NavigationMenuContent {
    home: string;
    websiteName: string;
    cryptoCurrency: string;
    candleChart: string;
    about: string;
}

export interface HomeContent {
    articleTitle: string;
    article: string;
}