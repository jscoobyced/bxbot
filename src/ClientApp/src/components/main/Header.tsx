import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { HeaderProps } from './Models';

export class Header extends React.Component<HeaderProps> {

    public constructor(props: HeaderProps) {
        super(props);
    }

    public render() {
        return <header>
            <img src='/images/wings.svg' className='logo'
                alt={this.props.navigationMenuContent.websiteName + ' logo'}
                title={this.props.navigationMenuContent.websiteName} />
            <span>{this.props.navigationMenuContent.websiteName}</span>
            <nav>
                <NavLink to={'/'}>
                    {this.props.navigationMenuContent.home}
                </NavLink>
                <a href='#' className='menu-space'>&nbsp;</a>
                <a href='#' className='dropdown'>{this.props.navigationMenuContent.cryptoCurrency}</a>
                <div className="dropdown-content">
                    <NavLink to={'/candle'}>
                        {this.props.navigationMenuContent.candleChart}
                    </NavLink>
                </div>
                <NavLink to={'/'}>
                    {this.props.navigationMenuContent.about}
                </NavLink>
            </nav>
        </header>;
    }
}