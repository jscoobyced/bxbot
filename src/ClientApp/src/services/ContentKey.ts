export class ContentKey {
    public static WebsiteName = 'WEBSITENAME';
}