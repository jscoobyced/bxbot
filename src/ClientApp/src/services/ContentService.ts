import { ContentKey } from './ContentKey';
import { AllContent, Language } from './Models';

export class ContentService {

    private content: Array<{ [key: string]: string; }> = [];

    public updateContent(language: Language): AllContent {
        this.content[language] = {};
        this.content[language][ContentKey.WebsiteName] = 'sawan.io';
        this.content[language][ContentKey.Home] = 'Home';
        this.content[language][ContentKey.About] = 'About';
        this.content[language][ContentKey.CryptoCurrency] = 'Cryptocurrency';
        this.content[language][ContentKey.CandleChart] = 'Candle Chart';
        const date = new Date().getTime() - 3 * 86400000;
        const displayDate = (new Date(date).toISOString().substring(0, 10));
        this.content[language][ContentKey.MainArticleTitle] = `Latest news - ${displayDate}`;
        this.content[language][ContentKey.MainArticle] = 'Grand opening of sawan.io.';

        return this.updateAllContent(language);
    }

    public defaultContent(): AllContent {
        return {
            navigationMenuContent: {
                about: '',
                candleChart: '',
                cryptoCurrency: '',
                home: '',
                websiteName: '',
            },
            homeContent: {
                article: '',
                articleTitle: ''
            }
        };

    }

    private updateAllContent(language: Language): AllContent {
        return {
            navigationMenuContent: {
                about: this.content[language][ContentKey.About],
                candleChart: this.content[language][ContentKey.CandleChart],
                cryptoCurrency: this.content[language][ContentKey.CryptoCurrency],
                home: this.content[language][ContentKey.Home],
                websiteName: this.content[language][ContentKey.WebsiteName],
            },
            homeContent: {
                article: this.content[language][ContentKey.MainArticle],
                articleTitle: this.content[language][ContentKey.MainArticleTitle]
            }
        };
    }
}
