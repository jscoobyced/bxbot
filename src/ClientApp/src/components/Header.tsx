import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { ContentKey } from '../services/ContentKey';
import { ContentService } from '../services/ContentService';

export class Header extends React.Component {

    public render() {
        const websiteName = ContentService.getContent(ContentKey.WebsiteName);
        return <header>
            <img src='/images/wings.svg' className='logo' alt={websiteName + ' logo'} title={websiteName} />
            <span>{websiteName}</span>
            <nav>
                <NavLink to={'/'}>
                    Home
                </NavLink>
                <a href='#' className='menu-space'>&nbsp;</a>
                <a href='#' className='dropdown'>Features</a>
                <div className="dropdown-content">
                <NavLink to={'/candle'}>
                        CandleChart
                    </NavLink>
                    <NavLink to={'/'}>
                        Stuff #1
                    </NavLink>
                    <NavLink to={'/candle'}>
                        Stuff #2
                    </NavLink>
                </div>
                <NavLink to={'/'}>
                    About
                </NavLink>
            </nav>
        </header>;
    }
}