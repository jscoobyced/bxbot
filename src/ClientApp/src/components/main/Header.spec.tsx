import { shallow } from 'enzyme';
import * as React from 'react';
import { Header } from './Header';
import { ContentService } from '../../services/ContentService';

test('Header component is unchanged.', () => {
    const navigationMenuContent = new ContentService().defaultContent().navigationMenuContent;
    const header = shallow(<Header navigationMenuContent={navigationMenuContent} />);

    expect(header).toMatchSnapshot();
});