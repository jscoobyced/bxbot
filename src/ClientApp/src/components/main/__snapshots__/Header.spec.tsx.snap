// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component is unchanged. 1`] = `
<header>
  <img
    alt=" logo"
    className="logo"
    src="/images/wings.svg"
    title=""
  />
  <span />
  <nav>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    />
    <a
      className="menu-space"
      href="#"
    >
      Â 
    </a>
    <a
      className="dropdown"
      href="#"
    />
    <div
      className="dropdown-content"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/candle"
      />
    </div>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    />
  </nav>
</header>
`;
